Sign session key hashes with a server-side secret (#80046)
metabase/metabase: src/metabase/cmd/resources/other-env-vars.md

Current cache backend. Dynamically rebindable primarily for test purposes.

> ### `MB_SESSION_HASH_SECRET_KEY`

Reviewer: Bikeshedding: I suggest renaming this to something more generic such as `MB_SECRET_KEY` in case we want to use this secret for signing other things. Common examples: Django uses `SECRET_KEY`, Rails uses `secret_key_base`, Laravel uses `APP_KEY`.

We also don't need to mention hashing in the name, since the fact that we use an HMAC is an implementation detail. We could have used e.g. a CMAC which does no hashing but is still valid as a signature

Author: done

Author: Renamed to `MB_SESSION_SECRET_KEY` in fc6992d. Went with the session-scoped name over `MB_SECRET_KEY`: the thread consensus was one secret per purpose (that's why we dropped the encryption-key fallback), and a generic app secret invites the same lifecycle coupling by the back door.

Type: string<br>
Default: `null`

> When set, session keys are stored in the application database as an HMAC signed with this secret, so a valid session cannot be created or used with database access alone. Requirement: minimum 16 characters. Setting or changing this value logs out all active sessions.

Reviewer: ```suggestion
When set, session keys are stored in the application database signed with this secret, so a valid session cannot be created or used with database access or guessing a valid session ID. Requirement: minimum 16 characters. Setting or changing this value logs out all active sessions.
```

We don't need to mention HMAC, that is an implementation detail. The signature is the important part. Also mentioned guessing as an attack vector, similar to https://github.com/metabase/metabase/pull/80046/changes#r3815560685

Author: Applied minus the guessing clause (same reasoning as the warning-text thread — signing doesn't prevent guessing, randomness does). Now reads "signed with this secret, so a valid session cannot be created or used with database access alone."

Author: done
